modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/CustomReflectionObjectHandler.java

```diff
@@ -41,6 +49,17 @@ public Object coerce(Object object) {
         }
     }
 
+    @Override
+    public Wrapper find(String name, List<Object> scopes) {
+        Wrapper wrapper = super.find(name, scopes);
+
+        if (detectMissingParams && wrapper instanceof MissingWrapper) {
+            throw new MustacheScriptEngine.InvalidParameterException("Parameter [" + name + "] is missing");
+        }
+
+        return wrapper;
+    }
```

Comment:
@jdconrad ^ this is the method of which I am most suspicious. Does this seem OK to you?

Looking at the mustache plugin code, it does seem ok, it will return a `MissingWrapper` if absolutely nothing is found.

Reply:
This seems fine to me as well as there's no behavioral change as long as `detectMissingParams` defaults to `false`.

Reply:
I did some change here so we can target only `ValueCode`.
The result is that we can now using conditional with missing parameters without throwing an exception (which is expected from my POV).

Here is the test I did add for it (other tests remain unchanged):

```
// do not throw when using a missing param in the conditional when DETECT_MISSING_PARAMS_OPTION option is set to true
        {
            String source = "{\"match\": { \"field\": \"{{#query_string}}{{.}}{{/query_string}}\" }";
            TemplateScript.Factory compiled = qe.compile(null, source, TemplateScript.CONTEXT, scriptOptions);
            Map<String, Object> params = Map.of();
            assertThat(compiled.newInstance(params).execute(), equalTo("{\"match\": { \"field\": \"\" }"));
        }
```